File: PokemonApp.Domain/Entities/Pokemon.cs
namespace PokemonApp.Domain.Entities
{
    public class Pokemon
    {
        public int Number { get; set; }

        public string Name { get; set; }
        
        public string Generation { get; set; }
        
        public int Height { get; set; }
        
        public int Weight { get; set; }
        
        public IEnumerable<string> Types { get; set; }
        
        public IEnumerable<Stat> Stats { get; set; }
        
        public IEnumerable<string> Moves { get; set; }
        
        public IEnumerable<string> Abilities { get; set; }
        
        public Evolution Evolution { get; set; }
        
        public string Image { get; set; }
    }
}

File: client-app/src/types/pokemon.ts
export interface Pokemon {
  number: number;
  name: string;
  generation: string;
  height: number;
  weight: number;
  types: string[];
  stats: { name: string; value: number }[];
  movesCount: number;
  moves: string[];
  abilities: string[];
  evolution: { from: string | null; to: string[] };
  image: string;
}

